### PR TITLE
rewrite healthcheck host headers - set hostname

### PR DESCRIPTION
**Pull request by ashleywang1**
BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/2794

### CODE REVIEW

**Comment by ashleywang1** [comment]
Here is how I tested to check that our healthchecks and requests were having their host headers replaced properly:

1) Start up 2 services that run 'mendhak/http-https-echo', a docker image that will log and echo back all the information in the request that is sent to it:
```
kind: Deployment
apiVersion: apps/v1
metadata:
  name: headers1
  namespace: default
  labels:
    name: headers1
spec:
  replicas: 1
  selector:
    matchLabels:
      task: headers1
  template:
    metadata:
      labels:
        task: headers1
    spec:
      containers:
        - name: log-and-echo-headers
          image: mendhak/http-https-echo
          ports:
            - containerPort: 80
---
kind: Deployment
apiVersion: apps/v1
metadata:
  name: headers2
  namespace: default
  labels:
    name: headers2
spec:
  replicas: 1
  selector:
    matchLabels:
      task: headers2
  template:
    metadata:
      labels:
        task: headers2
    spec:
      containers:
        - name: log-and-echo-headers
          image: mendhak/http-https-echo
          ports:
            - containerPort: 80
---
apiVersion: v1
kind: Service
metadata:
  name: headers1
  namespace: default
spec:
  ports:
    - name: http
      port: 80
  selector:
    task: headers1
---
apiVersion: v1
kind: Service
metadata:
  name: headers2
  namespace: default
spec:
  ports:
    - name: http
      port: 80
  selector:
    task: headers2
```

2) 
# look at `k get service -A` to get the cluster IPs of header1 and header2
# change the upstream hosts:
```
apiVersion: gloo.solo.io/v1
kind: Upstream
metadata:
  name: test-upstream
  namespace: gloo-system
spec:
  healthChecks:
  - healthyThreshold: 1
    httpHealthCheck:
      path: /healthcheck
    interval: 30s
    timeout: 10s
    unhealthyThreshold: 1
  static:
    hosts:
    - addr: 10.xxxxxx ## this may be different
      port: 80
    - addr: 10.xxxxxx ## this may be different
      port: 80
---
apiVersion: gateway.solo.io/v1
kind: VirtualService
metadata:
  name: test-vs
  namespace: gloo-system
spec:
  virtualHost:
    domains:
      - 'foo'
    routes:
      - matchers:
        - prefix: /
        routeAction:
          single:
            upstream:
              name: test-upstream
              namespace: gloo-system
        options:
          autoHostRewrite: true
``` 

Now, look at the logs for the header1 and header2 pods. You should see that the healthcheck requests (with the /healthcheck path) have the correct headers (e.g. the IP address of that service)

**Comment by ashleywang1** [comment]
/kick [Fail] Kube2e: gateway tests with virtual service with a mix of valid and invalid virtual services [BeforeEach] preserves the valid virtual services in envoy when a virtual service has been made invalid  flake https://storage.googleapis.com/solo-public-build-logs/logs.html?buildid=0302e088-8ba6-4d73-9824-a9e1cd1643d1

**Comment by ashleywang1** [comment]
/kick Consul e2e [It] works as expected test failure - new issue?

**Comment by ashleywang1** [comment]
/kick 
[Fail] Kube2e: Ingress [It] works  new flake ? https://console.cloud.google.com/cloud-build/builds/3f0156b9-2796-48a1-a032-0bb802ed7acc;step=17?project=solo-public

**Comment by ashleywang1** [comment]
/kick ERROR: build step 1 "gcr.io/cloud-builders/wget" failed: step exited with non-zero status: 255 https://storage.googleapis.com/solo-public-build-logs/logs.html?buildid=d75506f9-f56f-4d6b-9e0e-f6c06df186bb

**Comment by ashleywang1** [comment]
/kick [Fail] Knative RenderKnativeManifests [It] renders manifests for each knative component  test flake https://console.cloud.google.com/cloud-build/builds/912c409a-4fac-46ab-b0c9-7828ca9a4ee2;step=9?project=solo-public

**Comment by ashleywang1** [comment]
/kick 
ERROR: build step 1 "gcr.io/cloud-builders/wget" failed: step exited with non-zero status: 255 https://storage.googleapis.com/solo-public-build-logs/logs.html?buildid=5c9fcf27-9260-4ac5-817f-7876c42cd0c3

File: projects/gloo/pkg/plugins/aws/ec2/ec2_test.go [hunk 1]
```diff
@@ -150,6 +154,10 @@ var _ = Describe("Plugin", func() {
 						Namespace:   writeNamespace,
 						Annotations: map[string]string{InstanceIdAnnotationKey: "id1"},
 					},
+					Hostname: pubIp,
```
Comment (yuval-k) [inline projects/gloo/pkg/plugins/aws/ec2/ec2_test.go:157]: out of curiosity - what's the value of setting the hostname to an ip?

**Pull request closed without merging**